pkg/plugins/utils/cargo/main.go

```diff
@@ -0,0 +1,19 @@
+package cargo
+
+type InlineKeyChain struct {
+	// [A][S][C] Token specifies the cargo registry token to use for authentication.
+	Token string `yaml:",omitempty"`
+	// [A][S][C] HeaderFormat specifies the cargo registry header format to use for authentication (defaults to `Bearer`).
+	HeaderFormat string `yaml:"headerFormat,omitempty"`
+}
+
+type Registry struct {
```

Comment:
I like this approach but maybe, we miss a validation function that check that we didn't specified at the same time an URL, a rootdir, or a scmid.

Reply:
Would that be in a `func (s Spec) Validate() (bool)` function?

Reply:
I think I would create a function 
`func (r Registry) Validate() (bool)` here
Then call that function in the `New` function